### PR TITLE
Users references relation

### PR DESCRIPTION
**Pull request by briankung**
**Problem:** References weren't capturing data about who created them

**Solution:** 
- [x] Capture and backfill reference owner data
- [x] Mention user's username on their lists index page
- [x] Sort references by votes, then creation date instead of votes then publish date

### CODE REVIEW

File: app/assets/stylesheets/_crossref_search.scss [hunk 1]
```diff
@@ -18,6 +18,7 @@
   span.input-group-addon {
     background-color: $brand-primary;
     color: white;
+    cursor: pointer;
```
Comment (wsf246) [inline app/assets/stylesheets/_crossref_search.scss:21]: fancy

File: app/views/lists/index.html.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <h1>
-  <span>Lists</span>
+  <span><%= "#{@user.username}'s" if @user %> Lists</span>
```
Comment (wsf246) [inline app/views/lists/index.html.erb:2]: _thumbs up_

File: db/migrate/20161025201812_add_user_to_references.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserToReferences < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :references, :user, foreign_key: true
```
Comment (wsf246) [inline db/migrate/20161025201812_add_user_to_references.rb:3]: didn't know this was a thing.

**Comment by wsf246** [comment]
We should consider adding "added by xxxx" to each reference on a list, or maybe only if it wasn't added by the creator of the list.  Otherwise LGTM.

File: db/migrate/20161025201812_add_user_to_references.rb (hunk 3 above)
Comment (briankung) [reply db/migrate/20161025201812_add_user_to_references.rb:3]: This is generated by `rails generate migration AddUserToReferences user:references`

And yeah, you could just do an integer, too. I'm not sure what the `foreign_key: true` bit does

File: app/views/lists/index.html.erb (hunk 2 above)
Comment (briankung) [reply app/views/lists/index.html.erb:2]: If you hit colon : you can start typing and get a emoji selector 😆  thumbs up is : +1:

File: app/assets/stylesheets/_crossref_search.scss (hunk 1 above)
Comment (briankung) [reply app/assets/stylesheets/_crossref_search.scss:21]: Glad you liked it 👍

**Pull request merged**